src/test/java/org/openmrs/contrib/qaframework/automation/ClinicalVisitSteps.java

```diff
@@ -0,0 +1,263 @@
+	}
+
+	// User story: Add known allergies
+	@When("a user clicks on Allergies link from Patient dashboard page")
+	public void loadAllergiesPage() {
+		allergyPage = (AllergyPage) dashboardPage.clickOnAllergiesWidgetLink()
+				.waitForPage();
+	}
+
+	@Then("the system loads Allergies board page")
+	public void systemLoadsAllergiesPage() {
+		assertTrue(textExists("Allergies"));
+	}
+
+	@When("a user clicks Add Known Allergy button")
+	public void addKnownAllergy() {
+		addOrEditAllergyPage = allergyPage.clickOnAddNewAllergy();
+		addOrEditAllergyPage.enterDrug(DRUG_NAME);
+		addOrEditAllergyPage.drugId();
+		addOrEditAllergyPage.enterReaction(REACTION);
+		addOrEditAllergyPage.reactionId();
+		addOrEditAllergyPage.addAllergyNote("The effect is severe");
+	}
+
+	@And("a user clicks on save allergy button")
+	public void saveKnownAllergy() {
+		allergyPage = addOrEditAllergyPage.clickOnSaveAllergy();
+	}
+
+	@Then("the system adds known allergies into the allergies table")
+	public void systemAddsKnownAllergy() {
+		assertNotNull(addOrEditAllergyPage.getAllergiesList());
+		dashboardPage = addOrEditAllergyPage.clickReturn();
+	}
+
+	// User story: Add known condition
+	@When("a user clicks on Conditions link from Patient dashboard")
+	public void loadManageConditionsPage() {
+		conditionsPage = (ConditionsPage) dashboardPage
+				.clickOnConditionsWidgetLink().waitForPage();
+	}
+
+	@Then("the system loads Manage Conditions Page")
+	public void systemLoadsManageConditionsPage() {
+		assertTrue(conditionsPage.containsText("Conditions"));
+	}
+
+	@When("a user clicks on Add new condition")
+	public void userClicksAddNewCondition() {
+		conditionPage = (ConditionPage) conditionsPage.clickOnAddNewCondition()
+				.waitForPage();
+	}
+
+	@And("a user enters patient condition")
+	public void enterExistingCondition() {
+		conditionPage.typeInCondition(CONDITION);
+	}
+
+	@And("a user clicks on save condition button")
+	public void saveCondition() {
+		conditionPage.clickSave();
+	}
+
+	@Then("the system adds New Condition in Conditions table")
+	public void systemAddsCondition() {
+		assertNotNull(conditionsPage.getConditionsList());
+		dashboardPage = conditionsPage.clickReturn();
+	}
+
+	// User story: Attach supporting document
+	@When("a user clicks on Attachments link from patient visits dashboard")
+	public void loadAttachmentsPage() {
+		visitsDashboardPage = (PatientVisitsDashboardPage) dashboardPage
+				.goToRecentVisits();
+		attachmentsPage = (AttachmentsPage) dashboardPage.goToAttachmentsPage()
+				.waitForPage();
+	}
+
+	@Then("the system loads Attachments page")
+	public void systemloadsAttachmentsPage() {
+		assertTrue(textExists("Attachments"));
+	}
+
+	@When("a user attaches patient supporting file")
+	public void addSupportingFile() {
+		attachmentsPage.setFileUrl("/home/opensource/Documents/Form.pdf");
+		attachmentsPage.addAttachmentNote("Client medical history form");
+	}
+
```

Comment:
The user story that consumes `addSupportFile() `has been temporarily ignored not to run ci since we are currently missing a method in [uitestframework](https://github.com/openmrs/openmrs-contrib-uitestframework) that can  create a document file. Something like `createTestFille() `which can be deployed within the test for CI to pass. At the moment the method `setFileUrl()` uses a hard coded path which will passes locally but by default will fail with ci builds

Once we have the `creatTestFile()` function in the `uitestframework` repo, then we will do a little change in the test to deploy that file so that ci passes.